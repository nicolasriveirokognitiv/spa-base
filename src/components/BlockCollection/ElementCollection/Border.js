import React, { useState, useCallback } from 'react';
import PropTypes from 'prop-types';
import useTestMode from 'Hooks/useTestMode';
import { Switch } from 'antd';
import { elementCategories } from 'Constants/global';
import { toolPropTypes } from 'Constants/propTypes';
import BackgroundColor from './BackgroundColor';

function Border({ onChange, value, hasColorTool, testId }) {
    const [switchStatus, setSwitchStatus] = useState(value);
    const [border, setBorder] = useState({});

    const testProps = useTestMode(testId);

    const handleChildrenChange = useCallback((data, element) => {
        setBorder(prev => {
            const newState = { ...prev, [element]: data };
            onChange(newState);
            return newState;
        });
    }, [onChange]);

    const handleSwitchChange = useCallback(newSwitchStatus => {
        setSwitchStatus(newSwitchStatus);
        onChange(newSwitchStatus ? border : null);
    }, [border, onChange]);

    return (
        <>
            <Switch
                checked={switchStatus}
                onClick={handleSwitchChange}
                {...testProps}
            />
            {hasColorTool && <BackgroundColor label="Border color" propertyKey="borderColor" onChange={handleChildrenChange} value={border.borderColor} />}
        </>
    );
}

const borderPropTypes = {
    hasColorTool : PropTypes.bool,
    value        : PropTypes.bool,
};
Border.propTypes = { ...toolPropTypes, ...borderPropTypes };
Border.defaultProps = {
    onChange         : () => {},
    value            : false,
    label            : 'Border',
    propertyKey      : 'border',
    wrapperClassName : 'type-switch',
    category         : elementCategories.STYLE,
    hasColorTool     : false,
};

export default Border;
