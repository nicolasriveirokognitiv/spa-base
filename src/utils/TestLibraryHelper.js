/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import initStore from 'State/store';

const store = initStore();

const renderWithRedux = children => render(
    <Provider store={store}>
        {children}
    </Provider>,
);

export * from '@testing-library/react';
export { renderWithRedux, render };
