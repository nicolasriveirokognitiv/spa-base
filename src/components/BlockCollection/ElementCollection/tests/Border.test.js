import React from 'react';
import { render, fireEvent } from 'Utils/TestLibraryHelper';
import EditorContext from 'State/editorContext';
import Border from '../Border';

describe('Border', () => {
    let container;
    let props;

    beforeEach(() => {
        container = document.createElement('div');
        document.body.appendChild(container);
        props = {
            onChange : jest.fn(),
            value    : true,
        };
    });

    const renderBorder = p => render(
        <EditorContext.Provider value={{ isTestModeActive: true }}>
            <Border {...props} {...p} />
        </EditorContext.Provider>,
    );

    test('Should render Correctly', () => {
        expect(<Border />).toBeDefined();
    });

    test('Should match snapshot', async () => {
        const { asFragment } = render(<Border />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('Switch status should be the same in value', async () => {
        const { getByTestId } = renderBorder({ testId: 'border-switch' });
        const element = getByTestId('border-switch');
        expect(element.checked).toBeTruthy();
    });

    test('should call onChange prop on changes in input', () => {
        const { getByTestId } = renderBorder({ testId: 'border-switch' });
        fireEvent.click(getByTestId('border-switch'));
        expect(props.onChange).toHaveBeenCalled();
    });

    // TO DO test change color (handleChildrenChange)
});
