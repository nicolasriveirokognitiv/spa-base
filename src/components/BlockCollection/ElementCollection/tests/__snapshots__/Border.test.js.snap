// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Border Should match snapshot 1`] = `
<DocumentFragment>
  <button
    aria-checked="false"
    class="ant-switch"
    role="switch"
    type="button"
  >
    <span
      class="ant-switch-inner"
    />
  </button>
</DocumentFragment>
`;
